commit 412d05bff9076db6c7e7ffbc681138c08403796d
Author: Rabin Vincent <rabinv@axis.com>
Date:   Thu Apr 20 14:37:46 2017 -0700

    mm: prevent NR_ISOLATE_* stats from going negative
    
    commit fc280fe871449ead4bdbd1665fa52c7c01c64765 upstream.
    
    Commit 6afcf8ef0ca0 ("mm, compaction: fix NR_ISOLATED_* stats for pfn
    based migration") moved the dec_node_page_state() call (along with the
    page_is_file_cache() call) to after putback_lru_page().
    
    But page_is_file_cache() can change after putback_lru_page() is called,
    so it should be called before putback_lru_page(), as it was before that
    patch, to prevent NR_ISOLATE_* stats from going negative.
    
    Without this fix, non-CONFIG_SMP kernels end up hanging in the
    while(too_many_isolated()) { congestion_wait() } loop in
    shrink_active_list() due to the negative stats.
    
     Mem-Info:
      active_anon:32567 inactive_anon:121 isolated_anon:1
      active_file:6066 inactive_file:6639 isolated_file:4294967295
                                                        ^^^^^^^^^^
      unevictable:0 dirty:115 writeback:0 unstable:0
      slab_reclaimable:2086 slab_unreclaimable:3167
      mapped:3398 shmem:18366 pagetables:1145 bounce:0
      free:1798 free_pcp:13 free_cma:0
    
    Fixes: 6afcf8ef0ca0 ("mm, compaction: fix NR_ISOLATED_* stats for pfn based migration")
    Link: http://lkml.kernel.org/r/1492683865-27549-1-git-send-email-rabin.vincent@axis.com
    Signed-off-by: Rabin Vincent <rabinv@axis.com>
    Acked-by: Michal Hocko <mhocko@suse.com>
    Cc: Ming Ling <ming.ling@spreadtrum.com>
    Cc: Minchan Kim <minchan@kernel.org>
    Cc: Vlastimil Babka <vbabka@suse.cz>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1499_412d05bff9076db6c7e7ffbc681138c08403796d.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1499_412d05bff9076db6c7e7ffbc681138c08403796d.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
